dmg/compress.c

```diff
@@ -5,6 +5,22 @@
 #include <bzlib.h>
 #include "dmg/adc.h"
 
+#ifdef HAVE_LIBLZMA
+  #include <lzma.h>
+
+  static int lzmaDecompress(unsigned char* inBuffer, size_t inSize, unsigned char* outBuffer, size_t outBufSize, size_t *decompSize)
+  {
+    lzma_ret lret;
+    uint64_t memlimit = UINT64_MAX;
```

Comment:
This seems a bit high? Perhaps it should be set based on some multiple of system memory?

Reply:
Hmm. My understanding of this parameter is it's just for liblzma to check "is there enough memory?"; it won't actually make it changes to how it compresses/decompresses to fit under the limit. I guess it might be useful as a sanity check, but I don't really like the idea that we'd refuse to compress/decompress some data because eg: the user requested a high compression level.

Reply:
Maybe I'm misunderstanding what this is for, but according to https://tukaani.org/xz/liblzma-api/container_8h.html#aa58f237f6cea97ef0eb9bf5c37a3008d, this parameter controls "how much memory the decoder is allowed to allocate". Setting this to UINT64_MAX (9,223,372,036,854,775,807) seems to suggest that it would be allowed to attempt to allocate an implausible number of bytes?

I guess most tools don't impose memory limits though, come to think of it....